lib/model/folder.go

```diff
@@ -450,6 +450,41 @@ func (f *folder) scanSubdirs(subDirs []string) error {
 
 		batch.append(res.File)
 		changes++
+
+		snap.WithBlocksHash(res.File.BlocksHash, func(ifi db.FileIntf) bool {
+			fi := ifi.(protocol.FileInfo)
+
+			select {
+			case <-f.ctx.Done():
+				return false
+			default:
+			}
+
+			if f.ignores.Match(fi.Name).IsIgnored() {
+				return true
+			}
+
+			if !fi.BlocksEqual(res.File) || res.File.Size != fi.Size {
+				return true
+			}
+
+			if !osutil.IsDeleted(mtimefs, fi.Name) {
+				return true
+			}
+
+			nf := fi
+			nf.SetDeleted(f.shortID)
+			nf.LocalFlags = f.localFlags
+			if fi.ShouldConflict() {
+				// We do not want to override the global version with
+				// the deleted file. Setting to an empty version makes
+				// sure the file gets in sync on the following pull.
+				nf.Version = protocol.Vector{}
```

Comment:
What's going to happen here is that the other side will reject our delete as whatever it has is already better... So we could skip this stuff and just `if.ShouldConflict() { /* fuckit */ return true }` somewhere above?

Reply:
So this is just a copy of the code below, I guess if you think this should be changed this should be a separate PR?

Reply:
The purposes are different. In the existing case we send out the null-versioned delete to indicate that 1) we don't have the file 2) our info is out of date and should be ignored. The effect will be that we immediately pull the file... Here, we're looking to optimise the order of index updates, but this index update is pointless. I dunno. I'm not really sure when this would happen, the renamed-from file needs to have the Unsupported, Ignored or ReceiveOnlyChanged flags... I guess if we uningnore a file at precisely the same time we also rename it?

I guess the comment could say something about this being weird and unlikely but just matches the behavior of somewhere else...

Reply:
Good point: Essentially this rename detection doesn't have any effect on a receive-only folders, i.e. this entire new block could be skipped `if f.localFlags == FlagLocalReceiveOnly`. Having the `ShouldConflict` check just for safety anyway seems sensible. And similar to the ignores check etc. above I'd also not handle it here but just `return true`.

Reply:
Can someone give me pseudo-code what this should look like?

It feels like I'm in the united nations headquarters parking lot with all the flags around me.

Reply:
I was thinking of something like this (with the new function not being a required change, just makes it easiert o take about):

Move this entire block into a function, lets say `detectRenames`. Then have

```
	if f.localFlags != FlagLocalReceiveOnly {
		f.detectRenames(allTheArguments)
	}
```

In that function drop your current `ShouldConflict` condition, and add a new one between `ignores.Match` and `BlocksEqual`:

```
f fi.ShouldConflict() {
     // This should not happen, but even if it does: Conflicting changes are sent with a zero version vector, i.e. won't ever be needed/synced by remotes thus can't participate in renames.
    return true
}
```

Reply:
What's the point of the return value if its not used when using detectRenames?

Reply:
Pseudo-diff (fake indentation) without the `detectRenames` which is just niceness (and maybe caused the confusion), hopefully it makes clear what I mean:

```diff
--- lib/model/folder.go	2020-05-11 14:39:42.028608307 +0200
+++ fordiff.txt	2020-05-11 14:39:11.701112734 +0200
@@ -451,6 +451,7 @@
 		batch.append(res.File)
 		changes++
 
+		if f.localFlags != protocol.FlagLocalReceiveOnly {
 		snap.WithBlocksHash(res.File.BlocksHash, func(ifi db.FileIntf) bool {
 			fi := ifi.(protocol.FileInfo)
 
@@ -460,6 +461,10 @@
 			default:
 			}
 
+			if fi.ShouldConflict() {
+				return true
+			}
+
 			if f.ignores.Match(fi.Name).IsIgnored() {
 				return true
 			}
@@ -475,16 +480,11 @@
 			nf := fi
 			nf.SetDeleted(f.shortID)
 			nf.LocalFlags = f.localFlags
-			if fi.ShouldConflict() {
-				// We do not want to override the global version with
-				// the deleted file. Setting to an empty version makes
-				// sure the file gets in sync on the following pull.
-				nf.Version = protocol.Vector{}
-			}
 			batch.append(nf)
 			changes++
 			return false
 		})
+		}
 	}
 
 	if err := batch.flush(); err != nil {
```